addons/web/static/src/views/calendar/calendar_controller.js

```diff
@@ -110,6 +108,12 @@ export class CalendarController extends Component {
                 await this.model.unlinkRecords(ids);
             },
         };
+        this.archiveEnabled = "active" in this.props.fields
+            ? !this.props.fields.active.readonly
+            : "x_active" in this.props.fields
+            ? !this.props.fields.x_active.readonly
+            : false;
+        this.deleteRecordsWithConfirmation = useDeleteRecords(this.displayDialog);
```

Comment:
hey @aab-odoo can you take a look to the code ? 
i have some rks:
- i'm passing the dialog from here because calendar_controller already have its own method to display a dialog which is not the standard one
- _getResIds wasn't removed from dynamicList as the body generation is returned to that file as the hook only generates confirm function
- in tests, i was using RPCMock instead of makeServerError because the test passes but it's red as there's an error raised
https://runbot.odoo.com/runbot/build/79752320

i also have an error because i imported the hook inside dynamicList so i wanted to valid with you
- there is another error related to dialog title but i will fix it